src/randomeffectsterm.jl

```diff
@@ -23,9 +23,15 @@ function StatsModels.termvars(t::RandomEffectsTerm)
     vcat(StatsModels.termvars(t.lhs), StatsModels.termvars(t.rhs))
 end
 
-function StatsModels.apply_schema(t::FunctionTerm{typeof(|)}, schema::StatsModels.FullRank,
-        Mod::Type{<:MixedModel})
-    lhs, rhs = apply_schema.(t.args_parsed, Ref(schema), Mod)
+function StatsModels.apply_schema(t::FunctionTerm{typeof(|)},
+                                  schema::StatsModels.FullRank,
+                                  Mod::Type{<:MixedModel})
+    schema = StatsModels.FullRank(schema.schema)
+    lhs, rhs = t.args_parsed
+    if !StatsModels.hasintercept(lhs) && !StatsModels.omitsintercept(lhs)
+        lhs = InterceptTerm{true}() + lhs
```

Comment:
This is also much nicer than the originally proposed fix because the LHS stays on the left (even if the LHS of a `|` is essentially the RHS or a `~`.